src/include/mpi.h.in

```diff
@@ -895,7 +895,11 @@ typedef int (MPIX_Grequest_wait_function)(int, void **, double, MPI_Status *);
 #define MPIX_ERR_EAGAIN               MPICH_ERR_FIRST_MPIX+4 /* Operation could not be issued */
 #define MPIX_ERR_NOREQ                MPICH_ERR_FIRST_MPIX+5 /* Cannot allocate request */
 
-#define MPICH_ERR_LAST_MPIX           MPICH_ERR_FIRST_MPIX+5
+#define MPIX_ERR_PARTITION            MPICH_ERR_FIRST_MPIX+6 /* Invalid partition value */
+#define MPIX_ERR_PARTITIONS           MPICH_ERR_FIRST_MPIX+7 /* Invalid partitions value */
+#define MPIX_ERR_MISMATCH_SIZE        MPICH_ERR_FIRST_MPIX+8 /* Mismatch buffer size in partitioned pt2pt */
+
```

Comment:
I agree that the error codes might be a bit fine-grained. I am not very confident with the error code definition as no error code is standardized yet for partitioned comm. Thus, I was just listing all possible errors **explicitly mentioned** in the standard. For other **implementation-dependent** error, I treat them as internal error.

I am trying to find an existing code that might have a similar scenario.  However, what I see is that the existing codes actually follow a specific/fine-grained way. E.g., see below
```
#define MPI_ERR_INFO        28      /* */
#define MPI_ERR_INFO_KEY    29      /* */
#define MPI_ERR_INFO_VALUE  30      /* */
#define MPI_ERR_INFO_NOKEY  31      /* */
```

I list them as `MPIX_` because of two reasons (1) they are not yet standardized (2) no clear view how the error codes will be standardized. I am not sure how much impact it will be if we change the `MPIX_` error code to `MPI_` (or delete `MPIX_` error code) later once they are standardized in spec, but sounds like less impact than changing an `MPI_` error code.

Reply:
> I agree that the error codes might be a bit fine-grained. I am not very confident with the error code definition as no error code is standardized yet for partitioned comm. Thus, I was just listing all possible errors **explicitly mentioned** in the standard. For other **implementation-dependent** error, I treat them as internal error.
> 
> I am trying to find an existing code that might have a similar scenario. However, what I see is that the existing codes actually follow a specific/fine-grained way. E.g., see below
> 
> ```
> #define MPI_ERR_INFO        28      /* */
> #define MPI_ERR_INFO_KEY    29      /* */
> #define MPI_ERR_INFO_VALUE  30      /* */
> #define MPI_ERR_INFO_NOKEY  31      /* */
> ```
> 
> I list them as `MPIX_` because of two reasons (1) they are not yet standardized (2) no clear view how the error codes will be standardized. I am not sure how much impact it will be if we change the `MPIX_` error code to `MPI_` (or delete `MPIX_` error code) later once they are standardized in spec, but sounds like less impact than changing an `MPI_` error code.

I see that all MPI error class are explicitly listed in the standard, that makes me think that we should not add additiional error class unless it is listed in the standard.

The argument for not to use `MPIX_ERR_` class is of independent reason. `MPIX_` are extension, that in theory, we can remove without affecting the main functionality. Partitioned communication is the officical standard feature, so you can't remove those MPIX_ error codes as if they are extensions.

Can we do this PR without add extra error classes, for example,  use `MPI_ERR_OTHER` if necessary?

Reply:
Yea, we can use `MPI_ERR_OTHER` for now. Easier for us to change when the partitioned errors are well standardized.

Reply:
Will make the changes